# Esta carpeta esta orientada a entrenar distintos modelos para comprobar la efectividad del modelo propuesto en el paper

En específico este notebook es para separar los datos

In [1]:
import pandas as pd

In [4]:
data= pd.read_csv('../datos/train_poll_v1s1.csv')
data.describe()

,numpregunta,ai
count,460.000000,460.000000
mean,4.500000,0.500000
std,2.875408,0.500544
min,0.000000,0.000000
25%,2.000000,0.000000
50%,4.500000,0.500000
75%,7.000000,1.000000
max,9.000000,1.000000


In [5]:
#show the classes on column ai
data['ai'].value_counts()

ai
1    230
0    230
Name: count, dtype: int64

Como se tratan de 460 muestras con clases balanceadas, se separaran los datos en 70% entrenamiento y 30% validacion para que todos los modelos usen la misma separación

In [6]:
#Seleccionar los primeros 320
data_entrenamiento = data.iloc[:320, :]
data_entrenamiento['ai'].value_counts()

ai
1    166
0    154
Name: count, dtype: int64

In [ ]:
#Seleccionar los ultimos 138
data_validacion = data.iloc[:138, :]
data_validacion['ai'].value_counts()
# Guardar como csv
data_validacion.to_csv('validation.csv', index=False)

: 

Las subselecciones de datos no presentan clases completamente balanceadas, pero cercanas a una distribución equitativa así que se usarán para los experimentos

# Dataset pequeño (60 muestras aleatorias)

In [16]:
# Seleccionar 30 muestras de cada clase
train_small = data_entrenamiento.groupby('ai').apply(lambda x: x.sample(30, replace=False)).reset_index(drop=True)
train_small['ai'].value_counts()
# Guardar como .csv
train_small.to_csv('train_small.csv', index=False)

C:\Users\angel\AppData\Local\Temp\ipykernel_6856\845590649.py:2: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.
  train_small = data_entrenamiento.groupby('ai').apply(lambda x: x.sample(30, replace=False)).reset_index(drop=True)


# Dataset grande (300 muestras aleatorias)


In [17]:
# Seleccionar 150 muestras de cada clase
train_big = data_entrenamiento.groupby('ai').apply(lambda x: x.sample(150, replace=False)).reset_index(drop=True)
train_big['ai'].value_counts()
# Guardar como .csv
train_big.to_csv('train_big.csv', index=False)


C:\Users\angel\AppData\Local\Temp\ipykernel_6856\3987299611.py:2: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.
  train_big = data_entrenamiento.groupby('ai').apply(lambda x: x.sample(150, replace=False)).reset_index(drop=True)
